content/blog/esp32-bootstrapping/index.md

```diff
@@ -0,0 +1,366 @@
+Implementation of those additional resources depends on used build scenario.
+
+1. **Single image build** creates the image that can be processed by the ROM loader (aka Simpleboot)
+2. **Multi image build** consist of 2nd stage bootloader and one *or more* application images
+
+The bootloader options we are considering:
+- IDF bootloader
+- MCUboot
+
+You can refer to the table with list of features and boot scenarios for the Zephyr RTOS on ESP32 at the end of this article.
+
+### Simpleboot
+
+The default build scenario in Zephyr RTOS port of ESP32. The configuration option indicating the use of this build scenario:
+- `CONFIG_ESP_SIMPLE_BOOT`
+
+***NOTE:** Do not confuse Simple Boot with the `CONFIG_DIRECT_BOOT`, whis is native to newest RISC-V based SoCs, such as ESP32-C3.*
+
+```mermaid
+flowchart TD
+	ROM .->|load| FP0;
+	APP .->|load/map| FP0;
+	subgraph "ROM code";
+	ROM;
+	end;
+	subgraph "Simpleboot chain";
+	ROM ==> APP;
+	end;
+	subgraph "NV Storage";
+	FP0;
+	end;
+	ROM@{ shape: round, label: "**ROM loader**" };
+	APP@{ shape: round, label: "**Application** </br>*(ESP image format)*" };
+	FP0@{ shape: das, label: "**Boot&Application** </br> *(Can use entire flash)*</br>*DT_NODE_LABEL(boot_partition)*</br>*DT_NODE_BY_FIXED_PARTITION_LABEL(mcuboot)*" };
+```
+
+The default booting (and building) method employed in Zephyr RTOS for ESP32 SoCs is referred to as Simple Boot. This method generates a single application binary output, thereby streamlining the complexity of the process. Although it may lack certain advanced features, its appeal lies in its straightforwardness, offering an out-of-the-box solution for initiating any application. This simplicity makes it an accessible choice for developers seeking an uncomplicated development experience.
+
+
+### Bootloaders
+
+
+#### IDF bootloader
+
+***NOTE:*** *The IDF bootloader is deprecated in Zephyr RTOS but it is discussed here for the reference and because it has been used in early port of ESP32 on Zephyr RTOS*
+
+The IDF bootloader is the default 2nd stage bootloader for the ESP-IDF platform provided by Espressif. In the Zephyr RTOS it was used to load the application image (RAM load and flash map), but further functionalities were not ported so it acts as an accessible and simple loader.
+
+The reasons for removal of IDF bootloader from the Zephyr RTOS port:
+
+- building during each application build in the `ExternalProject` stage. Building multiple binaries in single build turn is not allowed in Zephyr RTOS.
+- introduction of MCUboot Zephyr port, which supports `--sysbuild`.
+
+
+#### MCUboot
+
+The MCUboot over the years becomes de-facto standard bootloader in IOT realm and Espressif Systems adopted it in several forms depending on used framework.
+
+There are two config options that indicates the usage of the MCUboot bootloader:
+- `CONFIG_BOOTLOADER_MCUBOOT` *(enabled when building the application loaded by the MCUboot)*
```

Comment:
Suggestion: explain a bit clearly what each define means. It can be still unclear what they are for starters.